### PR TITLE
Restyle spcomp.

### PR DESCRIPTION
**Pull request by dvander**
This restyles include/*.h and compiler/*.{h,cpp} in AMTL style.

### CODE REVIEW

File: compiler/amxdbg.h [hunk 1]
```diff
@@ -28,146 +28,150 @@
...
+    int16_t states PACKED;     /* number of entries in the "state" table */
 } PACKED AMX_DBG_HDR;
-#define AMX_DBG_MAGIC   0xf1ef
+#define AMX_DBG_MAGIC 0xf1ef
 
 typedef struct tagAMX_DBG_FILE {
-  ucell   address       PACKED; /* address in the code segment where generated code (for this file) starts */
-  char    name[1];           /* ASCII string, zero-terminated */
+    ucell address
+        PACKED;   /* address in the code segment where generated code (for this file) starts */
```
Comment (Fyren) [inline compiler/amxdbg.h]: Ergh.

File: compiler/assembler.cpp [hunk 2]
```diff
@@ -64,1326 +64,1322 @@ class CellWriter;
...
-  {135, "sysreq.n",   sIN_CSEG, parm2 },  /* version 9 (replaces SYSREQ.d from earlier version) */
-  {161, "tracker.pop.setheap", sIN_CSEG, parm0 },
-  {160, "tracker.push.c", sIN_CSEG, parm1 },
-  { 35, "xchg",       sIN_CSEG, parm0 },
-  { 83, "xor",        sIN_CSEG, parm0 },
-  { 91, "zero",       sIN_CSEG, parm1 },
-  { 90, "zero.alt",   sIN_CSEG, parm0 },
-  { 89, "zero.pri",   sIN_CSEG, parm0 },
-  { 92, "zero.s",     sIN_CSEG, parm1 },
+    /* node for "invalid instruction" */
```
Comment (Fyren) [inline compiler/assembler.cpp]: This is probably better before.

File: compiler/amxdbg.h (hunk 1 above)
Comment (dvander) [reply compiler/amxdbg.h]: Thanks, will fix this.

File: compiler/assembler.cpp (hunk 2 above)
Comment (dvander) [reply compiler/assembler.cpp]: Agree, I'll disable clang-format around this list.

File: compiler/expressions.h [hunk 3]
```diff
@@ -33,50 +33,48 @@ struct svalue;
...
+    int hier9(value *lval);
+    int hier8(value *lval);
+    int hier7(value *lval);
+    int hier6(value *lval);
+    int hier5(value *lval);
+    int hier4(value *lval);
+    int hier3(value *lval);
+    int hier2(value *lval);
+    int hier1(value *lval1);
+    int plnge(int *opstr, int opoff, HierFn hier, value *lval, const char *forcetag,
```
Comment (Fyren) [inline compiler/expressions.h]: I'm not sure what the deal is, but it seems like where the asterisk gets nestled by the formatter varies.  Here, it's with the variable name and in other places its with the type.

Example in `compiler/assember.h`: `void assemble(const char* outname, memfile_t* fin);`

Comment (dvander) [reply compiler/expressions.h]: I think I have to forcefully disable this in the clang-format config.

**Comment by dvander** [comment]
no more shrieks of horror, let's do this

**Pull request merged**